### ISSUE TITLE
Unable to SET struct's with encoding.BinaryMarshaler after v3.0.0

### ISSUE THREAD

**Issue by lantins** [mediocregopher/radix#83]
Hey,

I've got a problem where I'm unable to `radix.FlatCmd(nil, "SET", "myKey", myStruct)` where the struct implements the `encoding.BinaryMarshaler` interface.

This works with `v3.0.0` and the pre-modules codebase, but fails with newer versions like `v3.2.2`.

Hoping you can point me in the right direction.

---

Output when using `v3.0.0`:

```shell
$ go run main.go
phew, no errors
```

---

Output when using `v3.2.2`:

```shell
$ go run main.go
panic: read tcp 127.0.0.1:62981->127.0.0.1:6379: i/o timeout

goroutine 1 [running]:
main.main()
	/Users/lantins/devel/tmp/radix-v3-test/main.go:28 +0x264
exit status 2
```

---

Example code:

```golang
package main

import (
	"encoding/json"
	"fmt"

	radix "github.com/mediocregopher/radix/v3"
)

type StructWithMarshal struct {
	Foo string
	Bar string
}

func (s StructWithMarshal) MarshalBinary() ([]byte, error) {
	return json.Marshal(s)
}

func (s *StructWithMarshal) UnmarshalBinary(b []byte) error {
	return json.Unmarshal(b, &s)
}

func main() {
	pool, _ := radix.NewPool("tcp", "127.0.0.1:6379", 1)

	swm := &StructWithMarshal{Foo: "Hello", Bar: "World"}
	if err := pool.Do(radix.FlatCmd(nil, "SET", "test-key", swm)); err != nil {
		panic(err)
	}

	fmt.Println("phew, no errors")
}
```

**Comment by mediocregopher** [comment]
Hi @lantins , good catch and thanks for the detailed bug report. The problem came around in v3.1 when I implemented the ability to encode structs as if they were maps, but I didn't properly account for the case of the struct implementing one of the supported encoding interfaces. I just pushed a fix and tagged it `v3.2.3`, lemme how that works for you.

**Comment by lantins** [comment]
Tested last night, seems to be working 'as expected' for me with `v3.2.3`, thank you for the speedy resolution and for your fine work :)
